src/rt/cover.c

```diff
@@ -1164,6 +1239,42 @@ void x_cover_setup_toggle_cb(sig_shared_t *ss, int32_t tag)
    model_set_event_cb(m, s, fn, (void *)(uintptr_t)tag, false);
 }
 
+static void cover_state_cb(uint64_t now, rt_signal_t *s, rt_watch_t *w, void *user)
+{
+   // I-th enum literal is encoded in i-th tag from first tag, that corresponds to enum value.
+   int size = signal_size(s);
+   uint32_t tag_index = 0;
+
+   switch (size) {
+   case 1:
+      tag_index = *((uint8_t*)signal_value(s));
+      break;
+   case 2:
+      tag_index = *((uint16_t*)signal_value(s));
+      break;
+   case 4:
+      tag_index = *((uint32_t*)signal_value(s));
+      break;
+   default:
+      fatal("Unsupported signal size: %d\n", size);
+   }
+
+   uint32_t *mask = ((uint32_t*) user) + tag_index;
```

Comment:
See the earlier comment about passing the tag number rather than the pointer to the callback and then change this to `get_cover_counter(m, tag_index)`.